src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -629,132 +795,159 @@ public enum SupremeVictors implements LogicCardInfo {
+            text "60 damage. Flip 2 coins. This attack does 10 damage times the number of heads to each of your opponent's Benched Pokémon. (Don't apply Weakness and Resistance for Benched Pokémon.)"
             energyCost C, C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 60
+              flip 2, {
+                opp.bench.each {
+                  damage 10, it
+                }
+              }
             }
           }
-
         };
       case CHARIZARD_G_20:
-        return basic (this, hp:HP100, type:FIRE, retreatCost:3) {
+        return basic (this, hp:HP100, type:R, retreatCost:3) {
           weakness W
           resistance F, MINUS20
           move "Flame Jet", {
-            text "Flip a coin. If heads, this attack does 40 damage to 1 of your opponent’s Pokémon."
+            text "Flip a coin. If heads, this attack does 40 damage to 1 of your opponent's Pokémon. (Don't apply Weakness and Resistance for Benched Pokémon.)"
             energyCost R, C
             attackRequirement {}
             onAttack {
-              damage 0
+              flip {
+                damage 40, opp.all.select("Choose a Pokémon to deal 40 damage to")
+              }
             }
           }
           move "Heat Blast", {
             text "50 damage. "
             energyCost R, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 50
             }
           }
-
         };
       case CHIMECHO_21:
-        return basic (this, hp:HP070, type:PSYCHIC, retreatCost:1) {
+        return basic (this, hp:HP070, type:P, retreatCost:1) {
           weakness P, PLUS20
           move "Heal Bell", {
             text "Remove 3 damage counters from each of your Pokémon."
             energyCost C
-            attackRequirement {}
+            attackRequirement {
+              assert my.all.find{it.numberOfDamageCounters()} : "Your Pokémon are healthy"
+            }
             onAttack {
-              damage 0
+              my.all.each {
+                heal 30, it
+              }
             }
           }
           move "Super Psywave", {
-            text "Choose 1 of your opponent’s Pokémon. Count the amount of Energy attached to that Pokémon. Put"
+            text "Choose 1 of your opponent's Pokémon. Count the amount of Energy attached to that Pokémon. Put that many damage counters on the Pokémon."
             energyCost P
-            attackRequirement {}
+            attackRequirement {
+              assert opp.all.find{it.cards.energyCount(C)} : "None of your opponent's Pokémon have any Energy attached"
+            }
             onAttack {
-              damage 0
+              def tar = opp.all.findAll{it.numberOfDamageCounters()}.select("Choose 1 of your oppopnent's Pokémon")
+              directDamage 10*tar.numberOfDamageCounters(), tar
```

Comment:
This should be checking and multiplying by energyCount, not numberOfDamageCounters